src/cluster/src/check/render.rs

```diff
@@ -0,0 +1,191 @@
+use async_channel::Receiver;
+use crate::{CheckStatus, CheckResult, CheckResults, CheckFailed, CheckSuggestion};
+
+const ISSUE_URL: &str = "https://github.com/infinyon/fluvio/issues/new/choose";
+
+/// Renders individual checks as they occur over time
+pub async fn render_check_progress(progress: &mut Receiver<CheckResult>) -> CheckResults {
+    let mut check_results = vec![];
+    while let Ok(check_result) = progress.recv().await {
+        render_check_result(&check_result);
+        check_results.push(check_result);
+    }
+    check_results
+}
+
+/// Renders a slice of `CheckResults` all at once
+pub fn render_check_results<R: AsRef<[CheckResult]>>(check_results: R) {
+    let check_results = check_results.as_ref();
+    for check_result in check_results {
+        render_check_result(check_result);
+    }
+}
+
+/// Render a single check result
+pub fn render_check_result(check_result: &CheckResult) {
+    use colored::*;
+
+    match check_result {
+        Ok(status) => render_check_status(status),
+        Err(e) => {
+            // Print one layer of source error
+            let cause = match std::error::Error::source(e) {
+                Some(underlying) => format!(": {}", underlying),
+                None => "".to_string(),
+            };
+            let msg = format!("{}{}", e, cause);
+            println!("🚨 {} {}", "[ERROR]:".bold().red(), msg.red());
+        }
+    }
+}
+
+/// Render a slice of `CheckStatus`es all at once
+pub fn render_check_statuses<R: AsRef<[CheckStatus]>>(check_statuses: R) {
+    let check_statuses = check_statuses.as_ref();
+    for check_status in check_statuses {
+        render_check_status(check_status);
+    }
+}
+
+/// Render a single check status
+pub fn render_check_status(check_status: &CheckStatus) {
+    use colored::*;
+    use crate::CheckStatus::*;
+
+    match check_status {
+        Pass(success) => {
+            println!("✅ {} {}", "ok:".bold().green(), success.green());
+        }
+        Fail(e @ CheckFailed::AutoRecoverable(_)) => {
+            println!("❕ {} {}", "warning:".bold().yellow(), e);
+            println!(
+                "  💡 {} this may be fixed automatically during startup",
+                "note:".bold()
+            );
+        }
+        Fail(CheckFailed::AlreadyInstalled) => {
```

Comment:
Why is `AlreadyInstalled` part of CheckFailed.  Seems like `AutoRecoverable` and `AlreadyInstalled` doesn't belong together?

Reply:
This code was already like this, it's only showing up in the diff because I moved it. If we want to we can open another issue to talk about the organization of it though